### PR TITLE
fix: validate s3 response

### PR DESCRIPTION
**Pull request by kukhariev**
## Description

- catch and log errors with status 200
-  enable retries on part  upload errors
-  optimize part size

## Related issues
#5227

## Has this been tested?

- [ ] 👍 yes, I added tests to the test suite
- [x] 💭 no, because this PR is a draft and still needs work
- [ ] 🙅 no, because this PR does not update server code
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: server/lib/object-storage/shared/object-storage-helpers.ts [hunk 1]
```diff
@@ -31,14 +31,9 @@ async function storeObject (options: {
 
   logger.debug('Uploading file %s to %s%s in bucket %s', inputPath, bucketInfo.PREFIX, objectStorageKey, bucketInfo.BUCKET_NAME, lTags())
 
-  const stats = await stat(inputPath)
   const fileStream = createReadStream(inputPath)
 
-  if (stats.size > CONFIG.OBJECT_STORAGE.MAX_UPLOAD_PART) {
```
[the remark this answers is not in the document]
Comment (kukhariev) [reply server/lib/object-storage/shared/object-storage-helpers.ts:37]: [__uploadUsingPut used for small files](https://github.com/aws/aws-sdk-js-v3/blob/40e752a45cfe52385f9ca12ba50f592655afe305/lib/lib-storage/src/Upload.ts#L181-L184)

File: server/lib/object-storage/shared/object-storage-helpers.ts [hunk 2]
```diff
@@ -181,7 +152,13 @@ async function multiPartUpload (options: {
     params: input
   })
 
-  await parallelUploads3.done()
+  const response = (await parallelUploads3.done()) as CompleteMultipartUploadCommandOutput
+
+  if (!response.Key) {
+    const message = `Error uploading ${objectStorageKey} to bucket ${bucketInfo.BUCKET_NAME}`
+    logger.error(message, { response, ...lTags() })
+    throw new Error(message)
```
[the remark this answers is not in the document]
Comment (kukhariev) [reply server/lib/object-storage/shared/object-storage-helpers.ts:160]: maybe [`leavePartsOnError=true`](https://github.com/aws/aws-sdk-js-v3/issues/2311#issuecomment-939413928) will be enough

[the remark this answers is not in the document]
Comment (emansom) [reply server/lib/object-storage/shared/object-storage-helpers.ts:160]: @kukhariev Very confusing variable naming on @aws part. Maybe add a comment linking to [the code](https://github.com/aws/aws-sdk-js-v3/blob/v3.164.0/lib/lib-storage/src/Upload.ts#L274) for future reference.

File: config/default.yaml [hunk 3]
```diff
@@ -152,7 +152,7 @@ object_storage:
     secret_access_key: ''
 
   # Maximum amount to upload in one request to object storage
-  max_upload_part: 2GB
+  max_upload_part: 512MB
```
[the remark this answers is not in the document]
Comment (kukhariev) [reply config/default.yaml:155]: but 5MB parts very slow, we would have to send 10,000 PUT requests to upload a 50GB file and then reassemble all those 10,000 parts into one file.
And only 100 if the partSize is set to 500 megabytes.

How about 100 MB?

**Comment by Chocobozzz** [comment]
Thanks @kukhariev and @kontrollanten for the PR and the review. Really appreciated.

**Pull request merged**